plugins/Marketplace/vue/src/Marketplace/Marketplace.vue

```diff
@@ -23,251 +23,243 @@
-      required: true,
-    },
     pluginTypeOptions: {
-      type: [Object, Array],
+      type: Object,
       required: true,
     },
-    sort: {
+    defaultSort: {
       type: String,
       required: true,
     },
     pluginSortOptions: {
-      type: [Object, Array],
+      type: Object,
       required: true,
     },
-    pluginsToShow: {
-      type: Array,
+    numAvailablePluginsByType: {
+      type: Object,
       required: true,
     },
-    query: {
+    isValidConsumer: Boolean,
+    isSuperUser: Boolean,
+    isAutoUpdatePossible: Boolean,
+    isPluginsAdminEnabled: Boolean,
+    isMultiServerEnvironment: Boolean,
+    installNonce: {
       type: String,
-      default: '',
+      required: true,
     },
-    numAvailablePlugins: {
-      type: Number,
+    activateNonce: {
+      type: String,
+      required: true,
+    },
+    deactivateNonce: {
+      type: String,
+      required: true,
+    },
+    updateNonce: {
+      type: String,
       required: true,
     },
   },
   components: {
+    ContentBlock,
+    PluginList,
     Field,
   },
   data(): MarketplaceState {
     return {
-      pluginSort: this.sort,
-      pluginTypeFilter: this.pluginType,
-      searchQuery: this.query,
+      loading: false,
+      fetchRequest: null,
+      fetchRequestAbortController: null,
+      pluginSort: this.defaultSort,
+      pluginTypeFilter: 'plugins',
+      searchQuery: '',
+      pluginsToShow: [],
     };
   },
   mounted() {
     Matomo.postEvent('Marketplace.Marketplace.mounted', { element: this.$refs.root });
+    watch(() => MatomoUrl.hashParsed.value, () => {
+      this.updateValuesFromHash(false);
+    });
+
+    this.updateValuesFromHash(true);
   },
   unmounted() {
     Matomo.postEvent('Marketplace.Marketplace.unmounted', { element: this.$refs.root });
   },
-  created() {
-    const addCardClickHandler = (selector: string) => {
-      const $nodes = $(selector);
-      if (!$nodes || !$nodes.length) {
-        return;
-      }
-
-      $nodes.each((index, node) => {
-        const $card = $(node);
-
-        $card.off('click.cardClick');
-        $card.on('click.cardClick', (event) => {
-          // check if the target is a link or is a descendant of a link
-          // to skip direct clicks on links within the card, we want those honoured
-          if ($(event.target).closest('a').length) {
-            return;
-          }
+  methods: {
+    updateValuesFromHash(forceFetch: boolean) {
+      let doFetch = forceFetch;
 
-          const $titleLink = $card.find('a.card-title-link');
-          if ($titleLink) {
-            event.stopPropagation();
-            $titleLink.trigger('click');
-          }
-        });
-      });
-    };
+      const newSearchQuery = MatomoUrl.hashParsed.value.query as string | undefined;
+      const newPluginSort = MatomoUrl.hashParsed.value.sort as string | undefined;
+      const newPluginTypeFilter = MatomoUrl.hashParsed.value.pluginType as string | undefined;
```

Comment:
```suggestion
      const newSearchQuery = MatomoUrl.hashParsed.value.query as string || undefined;
      const newPluginSort = MatomoUrl.hashParsed.value.sort as string || undefined;
      const newPluginTypeFilter = MatomoUrl.hashParsed.value.pluginType as string || undefined;
```

Reply:
Ah right, that didn't even occur to me.
It might be best if it's always a string, so I'd go with the first option.